server/xpub-controller/manuscript.js

```diff
@@ -11,172 +11,54 @@ class Manuscript {
-      stream.on('end', () => {
-        resolve(Buffer.concat(chunks))
-        if (uploadedSize !== fileSize) {
-          logger.warn(
-            'Reported file size for manuscript is different than the actual file size',
-          )
-        }
-      })
-    })
-    fileEntity = await FileModel.find(fileId)
-    await fileEntity.updateStatus('UPLOADED')
-
-    logger.info(
-      `Manuscript Upload fileContents::end ${filename} | ${manuscriptId}`,
-    )
-
-    logger.info(`Manuscript Upload S3::start ${filename} | ${manuscriptId}`)
-
-    fileEntity = await FileModel.find(fileId)
-
-    try {
-      await this.storage.putContent(fileEntity, fileContents, {
-        size: fileSize,
-      })
-      await fileEntity.updateStatus('STORED')
-    } catch (err) {
-      logger.error(
-        `Manuscript was not uploaded to S3: ${err} | ${manuscriptId}`,
-      )
-      await fileEntity.updateStatus('CANCELLED')
-      await fileEntity.delete()
-      clearInterval(handle)
-      throw err
-    }
-
-    logger.info(`Manuscript Upload S3::end ${filename} | ${manuscriptId}`)
-
-    let title = ''
-    try {
-      // also send source file to conversion service
-      title = await this.scienceBeamApi.extractSemantics(
-        this.config,
-        fileContents,
-        filename,
-        mimeType,
-      )
-    } catch (error) {
-      let errorMessage = ''
-      if (error.error.code === 'ETIMEDOUT' || error.error.connect === false) {
-        errorMessage = 'Request to science beam timed out'
-      } else {
-        errorMessage = error.message
-      }
-      logger.warn('Manuscript conversion failed', {
-        error: errorMessage,
-        manuscriptId,
-        filename,
-      })
-    }
-
-    // After the length file operations above - now update the manuscript...
-    manuscript = await ManuscriptModel.find(manuscriptId, this.userId)
-    const oldFileIndex = manuscript.files.findIndex(
-      element => element.type === 'MANUSCRIPT_SOURCE',
-    )
-
-    logger.info(
-      `Manuscript Upload found index ${oldFileIndex} ${
-        fileEntity.filename
-      } | ${manuscriptId}`,
-    )
-
-    if (oldFileIndex >= 0) {
-      const oldFile = await FileModel.find(manuscript.files[oldFileIndex].id)
-      manuscript.files.splice(oldFileIndex, 1)
-      await oldFile.delete()
-    }
-
-    const pendingFileIndex = manuscript.files.findIndex(
-      element => element.type === 'MANUSCRIPT_SOURCE_PENDING',
+    const progress = FilesHelper.startFileProgress(
+      pubsub,
+      ON_UPLOAD_PROGRESS,
+      startedTime,
+      predictedTime,
+      manuscriptId,
     )
 
-    manuscript.files[pendingFileIndex].type = 'MANUSCRIPT_SOURCE'
-
-    logger.info(`Manuscript Upload Manuscript::save ${title} | ${manuscriptId}`)
-    manuscript.meta.title = title
-    await manuscript.save()
-
-    const sourceList = manuscript.files.filter(
-      f => f.type === 'MANUSCRIPT_SOURCE',
+    await this.manuscriptHelper.uploadManuscriptFile(
```

Comment:
What happens if this function throws ? ... this should be wrapped in a try-catch

Reply:
this will be introduced as part of the #1521